stan/math/prim/fun/eigenvalues.hpp

```diff
@@ -7,13 +7,14 @@
 namespace stan {
 namespace math {
 
-template <typename T>
-Eigen::Matrix<std::complex<T>, -1, 1> eigenvalues(
-    const Eigen::Matrix<T, -1, -1>& m) {
-  check_nonzero_size("eigenvalues", "m", m);
-  check_square("eigenvalues", "m", m);
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
+inline auto eigenvalues(const EigMat& m) {
+  using PlainMat = plain_type_t<EigMat>;
```

Comment:
[question]
Is there a reason to specify `PlainMat` rather than just using `.eval()`?

```cpp
auto& m_eval = m.eval();
```

Reply:
We use `PlainMat` later as a template argument to the eigen solver, but to be completely honest I was just using the style also used in eigenvalues_sym.hpp and elsewhere. @SteveBronder - is there a reason to prefer one to the other?

Reply:
Good point about needing to use it to parameterize the solver.  So reducing this to `auto` plus `eval` won't be enough.  So I'd just leave it like this.